packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryResetPassword/index.tsx

```diff
@@ -64,82 +112,236 @@ const AccountRecoveryResetPassword = ({
       },
     });
 
-  const alertSuccessAndNavigate = useCallback(() => {
-    const successCompletePwdReset = ftlMsgResolver.getMsg(
-      `${viewName}-success-alert`,
-      'Password set'
-    );
-    alertBar.success(successCompletePwdReset);
-    navigate('/reset_password_with_recovery_key_verified', { replace: true });
-  }, [alertBar, ftlMsgResolver, navigate]);
+  useEffect(() => {
+    if (state.contextError) {
+      alertInvalidContext(state.contextError);
+    } else if (!state.supportsRecovery) {
+      setBannerState(BannerState.Redirecting);
+      navigate(`/complete_reset_password?${urlSearchContext.toSearchQuery()}`);
+    }
+  }, [state, navigate, urlSearchContext]);
 
-  const sendNewLinkEmail = () => {
-    // TODO: Hook up functionality to send a new verification link
-  };
+  if (linkStatus === 'damaged') {
+    return <LinkDamaged {...{ linkType: 'reset-password' }} />;
+  }
 
-  const onSubmit = () => {
-    try {
-      // TODO: - completeAccountPasswordResetWithRecoveryKey
-      //       - logViewEvent('verification.success');
-      //       - metrics.logUserPreferences('account-recovery', false)
-      //       - logFlowEventOnce('recovery-key-consume.success', viewName);
-      alertSuccessAndNavigate();
-    } catch (e) {
-      // TODO: - if token expired, re-render and show LinkExpired
-      //       - metrics event for error
-      //       - error feedback in alertBar
-    }
-  };
+  if (linkStatus === 'expired') {
+    return (
+      <LinkExpired {...{ linkType: 'reset-password', resendLinkHandler }} />
+    );
+  }
 
   return (
-    <>
-      {linkStatus === 'valid' && (
-        <>
-          <CardHeader
-            headingText="Create new password"
-            headingTextFtlId="create-new-password-header"
-          />
-          <FtlMsg id="account-restored-success-message">
-            <p className="text-sm mb-4">
-              You have successfully restored your account using your account
-              recovery key. Create a new password to secure your data, and store
-              it in a safe location.
+    <AppLayout>
+      {BannerState.Redirecting === bannerState && (
```

Comment:
This is just an FYI, because I have a ticket filed for this (https://mozilla-hub.atlassian.net/browse/FXA-7019):

To match parity with the content-server, the banner component should go below the CardHeader. This causes some issues with the LinkExpired component, so in https://mozilla-hub.atlassian.net/browse/FXA-6891 (work in progress in parallel, rebased on your branch) I'll be moving the banner out of this page and into the LinkExpired component. This also reduces duplication because the banner messages are the same for all expired link contexts.

Reply:
I'll move them so there is less to do later.